src/dc_simplify.rs

```diff
@@ -205,142 +163,87 @@ impl Simplify {
                     }
                 }
                 // the incoming message might contain invalid UTF8
-                ret += &to_string_lossy(line);
+                ret += line;
                 content_lines_added += 1;
                 pending_linebreaks = 1i32
             }
         }
-        if self.is_cut_at_end && (!self.is_cut_at_begin || 0 != content_lines_added) {
+        if is_cut_at_end && (!is_cut_at_begin || 0 != content_lines_added) {
             ret += " [...]";
         }
-        dc_free_splitted_lines(lines);
 
-        ret.strdup()
+        ret
     }
 }
 
 /**
  * Tools
  */
-unsafe fn is_empty_line(buf: *const libc::c_char) -> bool {
-    /* force unsigned - otherwise the `> ' '` comparison will fail */
-    let mut p1: *const libc::c_uchar = buf as *const libc::c_uchar;
-    while 0 != *p1 {
-        if *p1 as libc::c_int > ' ' as i32 {
+fn is_empty_line(buf: &str) -> bool {
+    for c in buf.chars() {
+        if c > ' ' {
             return false;
         }
-        p1 = p1.offset(1isize)
     }
 
     true
 }
 
-unsafe fn is_quoted_headline(buf: *const libc::c_char) -> bool {
+fn is_quoted_headline(buf: &str) -> bool {
     /* This function may be called for the line _directly_ before a quote.
     The function checks if the line contains sth. like "On 01.02.2016, xy@z wrote:" in various languages.
     - Currently, we simply check if the last character is a ':'.
     - Checking for the existence of an email address may fail (headlines may show the user's name instead of the address) */
-    let buf_len: libc::c_int = strlen(buf) as libc::c_int;
-    if buf_len > 80i32 {
-        return false;
-    }
-    if buf_len > 0i32 && *buf.offset((buf_len - 1i32) as isize) as libc::c_int == ':' as i32 {
-        return true;
-    }
 
-    false
+    buf.len() <= 80 && buf.ends_with(':')
 }
 
-unsafe fn is_plain_quote(buf: *const libc::c_char) -> bool {
-    if *buf.offset(0isize) as libc::c_int == '>' as i32 {
-        return true;
-    }
-
-    false
+fn is_plain_quote(buf: &str) -> bool {
+    buf.starts_with(">")
 }
 
 #[cfg(test)]
 mod tests {
```

Comment:
can you add some tests for the freestanding functions please, want to make sure we increase coverage and they do what we expect them to do

Reply:
Ok